server/util/db/db.go

```diff
@@ -119,12 +121,80 @@ func maybeRunMigrations(dialect string, gdb *gorm.DB) error {
 	return nil
 }
 
-func openDB(dialect string, args ...interface{}) (*gorm.DB, error) {
-	gdb, err := gorm.Open(dialect, args...)
+// instrumentGORM adds GORM callbacks that populate query metrics.
+func instrumentGORM(gdb *gorm.DB) {
+	// Add callback that runs before other callbacks that records when the operation began.
+	// We use this to calculate how long a query takes to run.
+	recordStartTime := func(db *gorm.DB) {
+		if db.DryRun || db.Statement == nil {
+			return
+		}
+		db.Statement.Settings.Store(gormStmtStartTimeKey, time.Now())
+	}
+	gdb.Callback().Create().Before("*").Register("buildbuddy:record_op_start_time", recordStartTime)
```

Comment:
could "buildbuddy:record_op_start_time" be a const?

Reply:
Done.